feat: add solana program regular upgrade release doc template (#903)
axelarnetwork/axelar-contract-deployments: releases/solana/Programs-upgrade-<DATE>-template.md

- The upgrade authority keypair is available.
- You're upgrading using the same verifiable build process `solana-verify`.

## Program Upgrade Tracking

| Program     | Env                | From version | To version | From hash | To hash | ✅ Done |
| ----------- | ------------------ | ------------ | ---------- | --------- | ------- | ------ |
| Gateway     | `devnet-amplifier` |              |            |           |         |        |
| ITS         | `devnet-amplifier` |              |            |           |         |        |
| Gas Service | `devnet-amplifier` |              |            |           |         |        |
| Governance  | `devnet-amplifier` |              |            |           |         |        |
| Multicall   | `devnet-amplifier` |              |            |           |         |        |

Where `Env` can be:

* devnet-amplifier
* Stagenet
* Testnet
* Mainnet

## Prerequisites

1. **Build environment**

   ```bash
   export BASE_IMAGE="solanafoundation/solana-verifiable-build@sha256:979b09eef544de4502a92e28a724a8498a08e2fe506e8905b642e613760403d3"
   export ENV=<devnet-amplifier|stagenet|testnet|mainnet>

>    export CHAIN_ID=solana-2

Reviewer: dont hardcode the CHAIN_ID since this value is not true for all envs

Author: fixed at 3946a4fd02cbe9c7b4bbbc7bc6abadb7570fb0dc

2. **Build the updated binaries**

In the [programs repository](https://github.com/eigerco/solana-axelar) root, build only the programs you need to upgrade.

   ```bash
solana-verify build --base-image $BASE_IMAGE --library-name <library_name> -- --no-default-features --features $ENV

   ```

   Where `library_name` can be:

   * axelar_solana_gateway
   * axelar_solana_its
   * axelar_solana_gas_service
   * axelar_solana_governance
   * axelar_solana_multicall

## Upgrade programs

1. **Declare environment variables**

   ```bash
export PROGRAM_BYTECODE_PATH="solana-axelar/target/deploy/<program_name>.so"
export PROGRAM_ID=<PROGRAM_ID>

export UPGRADE_AUTHORITY_KEYPAIR_PATH=<path/to/upgrade_authority_keypair.json>
export COMMIT_HASH=$(git -C solana-axelar rev-parse HEAD)
   ```

   **Note**: `PROGRAM_BYTECODE_PATH` and `PROGRAM_ID` needs to be updated for each program that is going to be deployed.

2. **Set seolana CLI on the convenient cluster**

   ```bash
   solana config set --url <mainnet|devnet>
   ```
   note: We deploy all Axelar test environments in devnet
   

3. **Upgrade Programs**

There is a special CLI command that will get the program_id for you:

```bash

> ./solana/solana-axelar-cli upgrade --program <gateway|gas-service|governance|its> $PROGRAM_BYTECODE_PATH

Author: fixed at 4811321375e80246343aab13a6676f7ef7d8402b

```

## Verify

Verification is **only possible in mainnet**. If deploying for test environments you can skip this step.

```bash
solana-verify verify-from-repo --remote --base-image $BASE_IMAGE \
  --commit-hash $COMMIT_HASH \
  --program-id $PROGRAM_ID \
  https://github.com/eigerco/solana-axelar \
  -- --no-default-features --features $ENV
```

> ## **Post-Upgrade Checklist**

Reviewer 2: ```suggestion
## Post-Upgrade Checklist
```

# Solana Program upgrade tracking doc <Date> (this is template)

> This guide is for **upgrading previously deployed Solana programs** as part of the GMP/ITS v1.0.0 initial release ([1](./2025-07-GMP-v1.0.0.md), [2](./2025-07-ITS-v1.0.0.md) respectively). It assumes that:

Reviewer: ```suggestion
This guide is for **upgrading Solana programs** as part of the GMP/ITS v1.0.0 initial release ([1](./2025-07-GMP-v1.0.0.md), [2](./2025-07-ITS-v1.0.0.md) respectively). It assumes that:
```

Author: fixed at 73313e153ac9a7391c0ca6e624cecefe2f78685d

- You're upgrading using the same verifiable build process `solana-verify`.

## Program Upgrade Tracking

| Program     | Program ID                                    | Env                | From version | To version | From hash | To hash | ✅ Done |
| ----------- | --------------------------------------------- | ------------------ | ------------ | ---------- | --------- | ------- | ------ |
| Gateway     | `gtwi5T9x6rTWPtuuz6DA7ia1VmH8bdazm9QfDdi6DVp` | `devnet-amplifier` |              |            |           |         |        |
| ITS         | `itsqybuNsChBo3LgVhCWWnTJVJdoVTUJaodmqQcG6z7` | `devnet-amplifier` |              |            |           |         |        |
| Gas Service | `gasd4em72NAm7faq5dvjN5GkXE59dUkTThWmYDX95bK` | `devnet-amplifier` |              |            |           |         |        |
| Governance  | `govmXi41LqLpRpKUd79wvAh9MmpoMzXk7gG4Sqmucx9` | `devnet-amplifier` |              |            |           |         |        |
| Multicall   | `mce2hozrGNRHP5qxScDvYyZ1TzhiH8tLLKxwo8DDNQT` | `devnet-amplifier` |              |            |           |         |        |

Where `Env` can be:

* devnet-amplifier
* Stagenet
* Testnet

> * Mainnet

Reviewer: we should be getting values directly from the env.json files to avoid duplication

## Prerequisites

1. **Build environment**

   ```bash
   export BASE_IMAGE="solanafoundation/solana-verifiable-build@sha256:979b09eef544de4502a92e28a724a8498a08e2fe506e8905b642e613760403d3"
   export ENV=<devnet-amplifier|stagenet|testnet|mainnet>
   ```

2. **Build the updated binaries**

In the [programs repository](https://github.com/eigerco/solana-axelar) root, build only the programs you need to upgrade.

   ```bash
solana-verify build --base-image $BASE_IMAGE --library-name <library_name> -- --no-default-features --features $ENV

   ```

   Where `library_name` can be:

   * axelar_solana_gateway
   * axelar_solana_its
   * axelar_solana_gas_service
   * axelar_solana_governance
   * axelar_solana_multicall

## Upgrade programs

1. **Declare environment variables**

   ```bash
PROGRAM_BYTECODE_PATH="solana-axelar/target/deploy/<program_name>.so"
PROGRAM_ID=<PROGRAM_ID>

UPGRADE_AUTHORITY_KEYPAIR_PATH=<path/to/upgrade_authority_keypair.json>
COMMIT_HASH=$(git -C solana-axelar rev-parse HEAD)
   ```

>    **Note**: `PROGRAM_BYTECODE_PATH` and `PROGRAM_ID` needs to be updated for each program that is going tobe deployed.

Reviewer: ```suggestion
   **Note**: `PROGRAM_BYTECODE_PATH` and `PROGRAM_ID` needs to be updated for each program that is going to be deployed.
```

> 2. **Set seolana CLI on the covenient cluster**

Reviewer: ```suggestion
2. **Set solana CLI on the convenient cluster**
```

Author: fixed at https://github.com/axelarnetwork/axelar-contract-deployments/commit/73313e153ac9a7391c0ca6e624cecefe2f78685d

>    ```bash
>    solana config set --url <moniker>
>    ```
> 
>    Where `moniker` can be:
> 
>    * devnet (We deploy all Axelar test envs in devnet)
>    * mainnet

Reviewer: ```suggestion
   ```bash
   solana config set --url <mainnet|devnet>
   ```
```

Author: fixed at https://github.com/axelarnetwork/axelar-contract-deployments/commit/73313e153ac9a7391c0ca6e624cecefe2f78685d

3. **Upgrade Programs**

```bash
solana program deploy --program-id $PROGRAM_ID --upgrade-authority $UPGRADE_AUTHORITY_KEYPAIR_PATH $PROGRAM_PATH

> ```

Reviewer: this approach is a bit dangerous and prone to human error since there is no validation. My recommendation is to wrap this commands into a script that has validation to ensure we are deploying the right program. I would recommend passing the contract name as an input and actually fetching the programId from the env.json files .

Author: I think this is fixed at fe291aeee031999815bae05886a0acb786c88631 , @frenzox @isi8787 please review changes.

> ## Verify

Reviewer 2: Verification is only possible on Solana mainnet. I wonder if we should mention that here, leave as is or remove the verification step altogether. wdyt @isi8787?
